crates/miden-lib/asm/miden/account.masm

```diff
@@ -584,6 +584,9 @@ end
 #! - PROC_ROOT is the hash of the procedure to check.
 #! - was_called is 1 if the procedure was called, 0 otherwise.
 #!
+#! NOTE: This returns true only if the procedure invoked account-restricted kernel APIs (e.g.,
+#! `exec.faucet::mint`) which trigger `authenticate_and_track_procedure`. Procedures that execute
+#! only local MASM instructions will return false even if they were executed.
 #! Invocation: exec
 export.was_procedure_called
```

Comment:
A couple of nits:

- Let's add a newline before "Invocation: exec".
- Should we clarify that `true` is actually `1` to reduce potential ambiguity? i.e., "Returns 1 only if ... "